### PR TITLE
9399 reviewed filter

### PR DESCRIPTION
**Pull request by cooperka**
Main change:
- Added a "reviewed" search filter

Screen Shot 2019-05-13 at 14 56 59

Other major changes to be aware of:
- None

Related issues:
- Refactor to DRY up react components -- will do soon in https://redmine.sassafras.coop/issues/9624
- Secondary buttons have white text in ELMO theme -- upcoming PR to tackle this and other regressions from https://redmine.sassafras.coop/issues/9454

### CODE REVIEW

File: app/javascript/components/search/ReviewedFilter.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        {/* Note: `id` is required for checkbox label to be clickable. */}
+        <Form.Check
+          id="is-reviewed"
+          type="checkbox"
+          label={I18n.t('filter.is_reviewed')}
+          checked={isReviewed || ''}
+          onChange={handleChangeIsReviewed}
+        />
+
+        <div className="btn-apply-container">
```
Comment (smoyth) [inline app/javascript/components/search/ReviewedFilter.js]: I am still kind of uneasy about these buttons being duplicated on each one. It's like 15 lines of duplicated code. Could we not just make a subclass of Popover called like FilterPopover that contains this?

**Comment by smoyth** [comment]
Note with the button colors, it's only happening in the ELMO theme, if that helps.

File: app/javascript/components/search/ReviewedFilter.js (hunk 1 above)
Comment (cooperka) [reply app/javascript/components/search/ReviewedFilter.js]: The inner components for each popover are different enough that it's hard to unify the entire thing, but I'll at least make each individual component more generic. e.g.

```jsx
// QuestionFilter
<Popover
  className="filters-popover wide display-logic-container"
  id="form-filter"
>
  <ConditionSetFormField />

  <div className="btn-apply-container condition-margin">
    <Button
      className="btn-apply"
      onClick={onSubmit}
    >
      {I18n.t('common.apply')}
    </Button>
  </div>
</Popover>
```

vs.

```jsx
// ReviewedFilter
<Popover
  className="filters-popover"
  id="reviewed-filter"
>
  {/* Note: `id` is required for checkbox label to be clickable. */}
  <Form.Check
    id="is-reviewed"
    type="checkbox"
    label={I18n.t('filter.is_reviewed')}
    checked={isReviewed || ''}
    onChange={handleChangeIsReviewed}
  />

  <div className="btn-apply-container">
    <Button
      className="btn-clear btn-secondary"
      onClick={this.handleClearReviewed}
    >
      {I18n.t('common.clear')}
    </Button>
    <Button
      className="btn-apply"
      onClick={onSubmit}
    >
      {I18n.t('common.apply')}
    </Button>
  </div>
</Popover>
```

I can add common `FilterPopover`, `FilterPopoverButtonsContainer`, `ApplyButton`, `ClearButton`, etc. which will help.

**Comment by cooperka** [comment]
Planning to address button colors in separate PR with other related BS4 issues that fell through the cracks. Will add 3-way toggle here.

**Comment by cooperka** [comment]
Added toggle (screenshot now in OP). Other tasks you mentioned also now in OP and will be done as part of separate issues. Ready for final review.

**Pull request merged**